### PR TITLE
Deprecate Image in favor of ImageRef variable (VPC)

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Deprecate Image in favor of ImageRef variable (VPC)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Deprecate Image in favor of ImageRef variable (VPC)
```

### CODE REVIEW

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachines.yaml [hunk 1]
```diff
@@ -245,11 +245,22 @@ spec:
             description: IBMVPCMachineSpec defines the desired state of IBMVPCMachine.
             properties:
               image:
-                description: 'Image is the id of OS image which would be install on
-                  the instance. Example: r134-ed3f775f-ad7e-4e37-ae62-7199b4988b00
-                  TODO: allow user to specify a image name is much reasonable. Example:
-                  ibm-ubuntu-18-04-1-minimal-amd64-2'
+                description: 'Deprecated: Use ImageRef instead Image is the id of
```
Comment (Amulyam24) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachines.yaml:248]: small nit: this statement is not clear, probably should be: `'Deprecated: Use ImageRef instead. Image is the id of ...`

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachinetemplates.yaml [hunk 2]
```diff
@@ -209,11 +209,23 @@ spec:
                       of the machine.
                     properties:
                       image:
-                        description: 'Image is the id of OS image which would be install
-                          on the instance. Example: r134-ed3f775f-ad7e-4e37-ae62-7199b4988b00
-                          TODO: allow user to specify a image name is much reasonable.
-                          Example: ibm-ubuntu-18-04-1-minimal-amd64-2'
+                        description: 'Deprecated: Use ImageRef instead Image is the
```
Comment (Amulyam24) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachinetemplates.yaml:212]: same here as well

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachines.yaml (hunk 1 above)
Comment (Prajyot-Parab) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachines.yaml]: Fixed.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachinetemplates.yaml (hunk 2 above)
Comment (Prajyot-Parab) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_ibmvpcmachinetemplates.yaml]: Fixed.

**Comment by Prajyot-Parab** [comment]
@Amulyam24 Thanks for your feedback, necessary changes done. PTAL

**Review by Amulyam24 — APPROVED** [review]
/lgtm

**Comment by Amulyam24** [comment]
> @Amulyam24 Thanks for your feedback, necessary changes done. PTAL

Having both `Image` and `ImageRef` was a bit misleading earlier. Gives a better idea now 👍🏼

**Comment by mkumatag** [comment]
> > @Amulyam24 Thanks for your feedback, necessary changes done. PTAL
> 
> Having both `Image` and `ImageRef` was a bit misleading earlier. Gives a better idea now 👍🏼

But this breaks the backward compatibility, removing a field without even deprecating isn't a great idea.

**Comment by Amulyam24** [comment]
> > > @Amulyam24 Thanks for your feedback, necessary changes done. PTAL
> > 
> > 
> > Having both `Image` and `ImageRef` was a bit misleading earlier. Gives a better idea now 👍🏼
> 
> But this breaks the backward compatibility, removing a field without even deprecating isn't a great idea.

@mkumatag, the initial idea was to rename them, `Image` --> `ImageID` and `ImageRef` --> `Image`.
Any thoughts on it?

**Comment by mkumatag** [comment]
> > > > @Amulyam24 Thanks for your feedback, necessary changes done. PTAL
> > > 
> > > 
> > > Having both `Image` and `ImageRef` was a bit misleading earlier. Gives a better idea now 👍🏼
> > 
> > But this breaks the backward compatibility, removing a field without even deprecating isn't a great idea.
> 
> @mkumatag, the initial idea was to rename them, `Image` --> `ImageID` and `ImageRef` --> `Image`.
> Any thoughts on it?

How do you handle a scenario where the user deployed a cluster with some older version and want to update the version of capibm to the latest version with the above change?

**Comment by Amulyam24** [comment]
> > > > > @Amulyam24 Thanks for your feedback, necessary changes done. PTAL
> > > > 
> > > > 
> > > > Having both `Image` and `ImageRef` was a bit misleading earlier. Gives a better idea now 👍🏼
> > > 
> > > 
> > > But this breaks the backward compatibility, removing a field without even deprecating isn't a great idea.
> > 
> > 
> > @mkumatag, the initial idea was to rename them, `Image` --> `ImageID` and `ImageRef` --> `Image`.
> > Any thoughts on it?
> 
> How do you handle a scenario where the user deployed a cluster with some older version and want to update the version of capibm to the latest version with the above change?

Right, that would be tricky. Would be better to keep it as is then.

**Pull request closed without merging**